safehttp/plugins/csp/csp.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package csp
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"strings"
+
+	"github.com/google/go-safeweb/safehttp"
+)
+
+// Directive is the name of a single CSP directive.
+type Directive string
+
+const (
+	DirectiveScriptSrc Directive = "script-src"
+	DirectiveStyleSrc  Directive = "style-src"
+	DirectiveObjectSrc Directive = "object-src"
+	DirectiveBaseURI   Directive = "base-uri"
+	DirectiveReportURI Directive = "report-uri"
+)
+
+const (
+	ValueHTTPS         = "https:"
+	ValueHTTP          = "http:"
+	ValueUnsafeEval    = "'unsafe-eval'"
+	ValueUnsafeInline  = "'unsafe-inline'"
+	ValueNone          = "'none'"
+	ValueStrictDynamic = "'strict-dynamic'"
+)
+
+// PolicyDirective contains a single CSP directive.
+type PolicyDirective struct {
+	dir       Directive
+	vals      []string
+	nonces    bool
+	lastNonce string
+	// readRand is used for dependency injection in tests.
+	readRand func([]byte) (int, error)
+}
```

Comment:
Make this a struct with public fields, i.e.:

```
type PolicyDirective struct {
  Directive Directive
  Values []string
  AddNonce bool
}
```

Now, it seems like `lastNonce` and `readRand` don't have to belong to `PolicyDirective`.

Reply:
If you pursue this, you can remove `NewPolicyDirective` too.